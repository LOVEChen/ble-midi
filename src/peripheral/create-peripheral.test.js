/* eslint-env jest */
jest.mock('bleno/lib/bleno', () => {
  const bleno = jest.fn();
  bleno.on = jest.fn();
  bleno.__mockBleno__ = true;
  return () => bleno;
});

let createPeripheral;
let mockBleno;

beforeEach(() => {
  jest.resetAllMocks();
  const Bleno = require('bleno/lib/bleno');
  mockBleno = new Bleno();
  createPeripheral = require('./create-peripheral').default;
});

test('should createPeripheral(name)', () => {
  const name = jest.fn();
  const peripheral = createPeripheral(name);
  expect(mockBleno.mock).toMatchSnapshot();
  expect(mockBleno.on.mock).toMatchSnapshot();
  expect(peripheral.__mockBleno__).toEqual(true);
});

test('should createPeripheral with default name', () => {
  const peripheral = createPeripheral();
  expect(mockBleno.mock).toMatchSnapshot();
  expect(mockBleno.on.mock).toMatchSnapshot();
  expect(peripheral.__mockBleno__).toEqual(true);
});
