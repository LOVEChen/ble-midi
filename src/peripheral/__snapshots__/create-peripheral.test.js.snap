// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should createPeripheral with default name 1`] = `
Object {
  "calls": Array [],
  "instances": Array [],
}
`;

exports[`should createPeripheral with default name 2`] = `
Object {
  "calls": Array [
    Array [
      "stateChange",
      [Function],
    ],
    Array [
      "advertisingStart",
      [Function],
    ],
    Array [
      "platform",
      [Function],
    ],
    Array [
      "addressChange",
      [Function],
    ],
    Array [
      "advertisingStop",
      [Function],
    ],
    Array [
      "servicesSet",
      [Function],
    ],
    Array [
      "accept",
      [Function],
    ],
    Array [
      "mtuChange",
      [Function],
    ],
    Array [
      "disconnect",
      [Function],
    ],
    Array [
      "rssiUpdate",
      [Function],
    ],
  ],
  "instances": Array [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
}
`;

exports[`should createPeripheral(name) 1`] = `
Object {
  "calls": Array [],
  "instances": Array [],
}
`;

exports[`should createPeripheral(name) 2`] = `
Object {
  "calls": Array [
    Array [
      "stateChange",
      [Function],
    ],
    Array [
      "advertisingStart",
      [Function],
    ],
    Array [
      "platform",
      [Function],
    ],
    Array [
      "addressChange",
      [Function],
    ],
    Array [
      "advertisingStop",
      [Function],
    ],
    Array [
      "servicesSet",
      [Function],
    ],
    Array [
      "accept",
      [Function],
    ],
    Array [
      "mtuChange",
      [Function],
    ],
    Array [
      "disconnect",
      [Function],
    ],
    Array [
      "rssiUpdate",
      [Function],
    ],
  ],
  "instances": Array [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
}
`;

exports[`should createPeripheral(name, onIncomingPacket) 1`] = `
Object {
  "calls": Array [],
  "instances": Array [],
}
`;

exports[`should createPeripheral(name, onIncomingPacket) 2`] = `
Object {
  "calls": Array [
    Array [
      "stateChange",
      [Function],
    ],
    Array [
      "advertisingStart",
      [Function],
    ],
    Array [
      "platform",
      [Function],
    ],
    Array [
      "addressChange",
      [Function],
    ],
    Array [
      "advertisingStop",
      [Function],
    ],
    Array [
      "servicesSet",
      [Function],
    ],
    Array [
      "accept",
      [Function],
    ],
    Array [
      "mtuChange",
      [Function],
    ],
    Array [
      "disconnect",
      [Function],
    ],
    Array [
      "rssiUpdate",
      [Function],
    ],
  ],
  "instances": Array [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
}
`;
